session.go

```diff
@@ -506,6 +508,14 @@ func (s *Session) recvLoop() error {
 			return err
 		}
 
+		// Reset the keepalive timer every time we receive data.
+		// There's no reason to keepalive if we're active. Worse, if the
+		// peer is busy sending us stuff, the pong might get stuck
+		// behind a bunch of data.
+		if s.keepaliveTimer != nil {
+			s.keepaliveTimer.Reset(s.config.KeepAliveInterval)
```

Comment:
This is the bugfix.

Reply:
Don't you need to check if the timer has already been read as described in https://golang.org/pkg/time/#Timer.Reset?

Reply:
this one gets me every time too -- the API for resetting timers is really poor.

Reply:
Not in this case. Reset actually stops it internally.

You have to stop it first and drain the channel if you want to make sure that the previous timer doesn't fire after you reset it but before new timeout elapses. However:

1. We don't _really_ care in this case.
2. This isn't a normal timer, it's an AfterFunc and doesn't have a channel to drain. There's no way to reliably stop an AfterFunc because the function is called asynchronously. Note: trying to drain a channel of a timer created by `AfterFunc` will just block forever.